src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -3374,7 +3378,9 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
                   // the overloaded type into a single function type from which `typedFunction`
                   // can derive the argument type for `x` in the function literal above
                   val argPt =
-                    if (argPtAlts.nonEmpty && treeInfo.isFunctionMissingParamType(tree)) functionProto(argPtAlts)
+                    if (argPtAlts.isEmpty) WildcardType
+                    else if (treeInfo.isFunctionMissingParamType(tree)) functionProto(argPtAlts)
+                    else if (treeInfo.isPartialFunctionMissingParamType(tree)) partialFunctionProto(argPtAlts)
```

Comment:
Looks like a non-trivial problem. Perhaps extract the argument types from `fun` after `inferMethodAlternative` (is there a simple way to do that?) and then rebuild `args1` with `Function1` types in place of `PartialFunction` where it is sufficient?